cudf/columnops.py

```diff
@@ -222,6 +222,23 @@ def as_column(arbitrary):
                           "be typecast to a Date64 value", UserWarning)
             arbitrary = arbitrary.cast(pa.date64())
             data = as_column(arbitrary)
+        elif isinstance(arbitrary, pa.BooleanArray):
+            # Arrow uses 1 bit per value while we use int8
+            dtype = np.dtype(np.bool)
+            arbitrary = arbitrary.cast(pa.int8())
+            if arbitrary.buffers()[0]:
```

Comment:
Perhaps add a bit more documentation to the logic here -- might be hard for someone else (or yourself in a year) to maintain.  What is in `arbitrary.buffers()[0]` and `[1]`?

It's unclear where the 1 bit per value is being compacted into a int8; perhaps explain that.

Reply:
It's not being compacted, each bit is being expanded to an int8 since that's what we use for booleans.

The subsequent lines show clear that buffers()[0] is the mask buffer and buffers()[1] is the data buffer, via variables `pamask` and `padata`.